src/AlgebraicGeometry/Surfaces/EnriquesAut.jl

```diff
@@ -0,0 +1,908 @@
+################################################################################
+# Computation of Automorphism groups of Enriques surfaces following
+# https://doi.org/10.1307/mmj/20195769
+# and
+# http://www.arxiv.org/abs/1909.10813
+# Initial Version by Simon Brandhorst
+################################################################################
+
+@doc raw"""
+    EnriquesBorcherdsCtx
+
+Holds data for computing invariants of families of Enriques surfaces with fixed root 
+invariants, in particular the automorphism group of a general element. 
+
+The setting is the following:
+Let ``Y`` be an Enriques surface over an algebraically closed field of characteristic not ``2``
+and ``π : X → Y`` its universal covering K3 surface. We denote by ``S_Y`` and ``S_X`` the 
+numerical lattices of ``Y`` and ``X``.
+
+Most importantly the type stores a chain of lattices
+
+```math
+S_Y(2) \subseteq S_X \subseteq L_{1,25}
+```
+
+where ``L_{1,15}`` is an even unimodular lattice of signature $(1,25)``. 
+```
+"""
+mutable struct EnriquesBorcherdsCtx
+  # SY < SX < L26
+  L26::ZZLat
+  SX::ZZLat # numerical lattice of X
+  SY::ZZLat # numerical lattice of Y with form rescaled by 2
+  # the following are given with respect to the basis of SY
+  initial_walls::Vector{ZZMatrix}
+  initial_rays::Vector{ZZMatrix}
+  initial_isotropic_rays::Vector{ZZMatrix}
+  # automorphisms of the initial chamber
+  # written w.r.t. the basis of SY
+  initial_automorphisms::Vector{ZZMatrix}
+  initial_automorphisms_mod2::Vector{FqMatrix}  # same order as above but mod 2
+  initial_chamber::K3Chamber   # a non-degenerate L26/SX chamber
+  # whether a -2 vector defines an outer wall is a mod 2 condition
+  roots_mod2::Set{FqMatrix}  # mod 2 images of splitting roots
+  gramSY::ZZMatrix  # to avoid coercions
+  gramSX::ZZMatrix  # to avoid coercions
+  membership_test
+  # needed for the membership test
+  # the following are in SY/2SY w.r.t the basis of SY mod 2
+  Dplus_perp::FqMatrix
+  Dplus::FqMatrix
+  imgs_mod2::Set{FqMatrix}
+  # w.r.t the basis given by Dplus
+  Gplus
+  Gplus_mat::MatrixGroup{FqFieldElem, FqMatrix}
+  function EnriquesBorcherdsCtx()
+    return new()
+  end
+
+  volume_index::ZZRingElem
+  orderGbar::ZZRingElem
```

Comment:
```suggestion
  volume_index::ZZRingElem
  orderGbar::ZZRingElem
  
  function EnriquesBorcherdsCtx()
    return new()
  end
```
keep all fields at the top for better readability